src/backend/downloadmanager/utils.ts

```diff
@@ -0,0 +1,108 @@
+import { BrowserWindow } from 'electron'
+import { logError, logInfo, LogPrefix, logWarning } from '../logger/logger'
+import { getGame, isEpicServiceOffline, notify } from '../utils'
+import { InstallParams } from 'common/types'
+import i18next from 'i18next'
+import { showErrorBoxModalAuto } from '../dialog/dialog'
+import { isOnline } from '../online_monitor'
+
+async function installQueueElement(
+  mainWindow: BrowserWindow,
+  params: InstallParams
+) {
+  const {
+    appName,
+    path,
+    installDlcs,
+    sdlList = [],
+    runner,
+    installLanguage,
+    platformToInstall
+  } = params
+  const game = getGame(appName, runner)
+  const { title } = game.getGameInfo()
+
+  if (!isOnline()) {
+    logWarning(`App offline, skipping install for game '${title}'.`, {
+      prefix: LogPrefix.Backend
+    })
+    return { status: 'error' }
+  }
+
+  const epicOffline = await isEpicServiceOffline()
+  if (epicOffline && runner === 'legendary') {
```

Comment:
maybe move this like this:

```ts
if (runner === 'legendary') {
  const epicOffline = await isEpic....
  if (epicOffile) {
    ...
  }
}
```
so we don't check if the services are online for gog games

Reply:
Makes sense 👍🏽